src/element-internals.ts

```diff
@@ -166,10 +166,14 @@ export class ElementInternals implements IElementInternals {
     }
     validationAnchorMap.set(this, anchor);
     const validity = validityMap.get(this);
-    if (Object.keys(validityChanges).length === 0) {
+    const validityChangesObj = {};
+    for (const key in validityChanges) {
+      validityChangesObj[key] = validityChanges[key];
```

Comment:
Would it be more efficient to jus run `{...other}`?

Reply:
Sorry should have been more clear about this in the PR description, spread syntax does not seem to work on the native `ValidityState` object, it yields an empty object.

Reply:
This is a one-liner you can test in the browser console: `({ ...document.createElement("input").validity })`

Yields an empty object in Chrome, Firefox and Safari

Reply:
Good call.